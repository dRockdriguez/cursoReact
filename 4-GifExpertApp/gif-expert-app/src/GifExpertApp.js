import React, { useState } from "react";
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";
import 'animate.css';

const GifExpertApp = () => {
  const [categories, setCategories] = useState(["One Punch"]);

  return (
    <>
      <h2 className="animate__animated animate__bounce">GifExpertApp</h2>
      <hr />
      <AddCategory setCategories={setCategories} />
      <ul>
        {categories.map((el) => (
          <GifGrid 
          key={`${el}${Date.now()}`}
          category={el}></GifGrid>
        ))}
      </ul>
    </>
  );
};

export default GifExpertApp;
