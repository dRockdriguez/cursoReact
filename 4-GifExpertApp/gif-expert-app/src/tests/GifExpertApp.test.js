import React from "react";
import GifExpertApp from "../GifExpertApp";
import { shallow } from "enzyme";

describe("GifExpertApp", () => {
  let wrapper = shallow(<GifExpertApp />);

  beforeEach(() => {
    wrapper = shallow(<GifExpertApp />);
  });

  test("Should render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Should render the categories list", () => {
    const categories = ['cat1', 'cat2'];
    wrapper = shallow(<GifExpertApp  defaultCategories={categories}/>);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
