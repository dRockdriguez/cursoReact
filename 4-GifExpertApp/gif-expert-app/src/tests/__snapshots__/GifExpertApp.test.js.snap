// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp Should render the categories list 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__bounce"
  >
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ul>
    <GifGrid
      category="cat1"
      key="cat11645716702637"
    />
    <GifGrid
      category="cat2"
      key="cat21645716702637"
    />
  </ul>
</Fragment>
`;

exports[`GifExpertApp Should render the component 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__bounce"
  >
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ul />
</Fragment>
`;
